server/game/core/ongoingEffect/OngoingEffectEngine.ts

```diff
@@ -93,12 +95,25 @@ export class OngoingEffectEngine {
 
     public removeLastingEffects(card: OngoingEffectSource) {
         this.unapplyAndRemove(
-            (effect) =>
-                effect.matchTarget === card &&
-                effect.duration !== Duration.Persistent
+            (effect) => this.shouldRemove(card, effect)
```

Comment:
If this is the only place `shouldRemove()` is called we could maybe just make it a code block here? Not a strong preference, I think it works equally fine either way

Reply:
I was thinking it may need to be reused elsewhere like where the other check for the limit is. Do you think we could use this method in place of that if we move it down after the increment like you proposed?